### PR TITLE
[#1] [#2] Implement Splash screens

### PR DESCRIPTION
**Pull request by chornerman**
#1, #2

## What happened 👀

- [x] Add Splash screen using [flutter_native_splash](https://pub.dev/packages/flutter_native_splash) library

## Insight 📝

- The Splash screen from the design [can only work on iOS and Android below version 12](https://pub.dev/packages/flutter_native_splash). Android 12+ devices will display the default one with the darker background color instead

## Proof Of Work 📹

https://user-images.githubusercontent.com/13492460/209579243-5271074d-699c-4468-83a0-3927f7701a38.mov

### CODE REVIEW

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -12,7 +11,6 @@
             android:hardwareAccelerated="true"
             android:label="@string/app_name"
             android:launchMode="singleTop"
-            android:screenOrientation="portrait"
```
Comment (luongvo) [inline android/app/src/main/AndroidManifest.xml:15]: I think the survey app does not support landscape orientation 🤔

File: android/app/src/main/res/values-night-v31/styles.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <!-- Theme applied to the Android Window while the process is starting when the OS's Dark Mode setting is off -->
+    <style name="LaunchTheme" parent="@android:style/Theme.Black.NoTitleBar">
+        <item name="android:forceDarkAllowed">false</item>
+        <item name="android:windowFullscreen">false</item>
+        <item name="android:windowLayoutInDisplayCutoutMode">shortEdges</item>
+        <item name="android:windowSplashScreenBackground">#15151A</item>
+    </style>
```
Comment (luongvo) [inline android/app/src/main/res/values-night-v31/styles.xml:9]: I guess this is generated code by `flutter_native_splash` 💭

File: android/app/src/main/res/values-v31/styles.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    </style>
+    <!-- Theme applied to the Android Window as soon as the process has started.
+         This theme determines the color of the Android Window while your
+         Flutter UI initializes, as well as behind your Flutter UI while its
+         running.
+         
+         This Theme is only used starting with V2 of Flutter's Android embedding. -->
+    <style name="NormalTheme" parent="@android:style/Theme.Light.NoTitleBar">
+        <item name="android:windowBackground">?android:colorBackground</item>
+    </style>
```
Comment (luongvo) [inline android/app/src/main/res/values-v31/styles.xml:18]: same, this style 💭

**Review by luongvo — COMMENTED** [review]
Not sure why do we need to duplicate the res into `v21`? Do you need to make any different between the default in drawable and v21?

image

I guess this is generated code by flutter_native_splash 💭

**Comment by chornerman** [comment]
@luongvo Regarding the themes and the new folder. Should we get rid of it? I'm not sure if doing so will break the correct behavior of the library since they're all generated files 🤔

**Comment by luongvo** [comment]
> @luongvo Regarding the themes and the new folder. Should we get rid of it? I'm not sure if doing so will break the correct behavior of the library since they're all generated files 🤔

Yeah, if all of them came from generated code by the lib, we can ignore them.

**Review by luongvo — APPROVED** [review]

**Review by markgravity — APPROVED** [review]

**Pull request merged**